src/main/java/com/uid2/admin/audit/QLDBInit.java

```diff
@@ -1,350 +1,134 @@
+                txn.execute("CREATE TABLE " + qldbTableName);
+            });
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
 
-    private static void addSaltModels(Collection<Collection<OperationModel>> models, RotatingSaltProvider saltProvider) {
+    private static void createIndices() { //creates indices on itemType and itemKey. Assumes they don't already exist.
         try {
-            List<RotatingSaltProvider.SaltSnapshot> snapshots = saltProvider.getSnapshots();
-            RotatingSaltProvider.SaltSnapshot lastSnapshot = snapshots.get(snapshots.size() - 1);
-
-            JsonObject jo = new JsonObject();
-            jo.put("effective", lastSnapshot.getEffective().toEpochMilli());
-            jo.put("expires", lastSnapshot.getExpires().toEpochMilli());
-            jo.put("salts_count", lastSnapshot.getAllRotatingSalts().length);
-            jo.put("min_last_updated", Arrays.stream(lastSnapshot.getAllRotatingSalts())
-                    .map(SaltEntry::getLastUpdated)
-                    .min(Long::compare).orElse(null));
-            jo.put("max_last_updated", Arrays.stream(lastSnapshot.getAllRotatingSalts())
-                    .map(SaltEntry::getLastUpdated)
-                    .max(Long::compare).orElse(null));
-
-            Collection<OperationModel> newModels = new HashSet<>();
-            newModels.add(new OperationModel(Type.SALT, "singleton", null,
-                    DigestUtils.sha256Hex(jo.toString()), null));
-            models.add(newModels);
+            qldbDriver.execute(txn -> {
+                txn.execute("CREATE INDEX ON " + qldbTableName + "(itemType)");
+                txn.execute("CREATE INDEX ON " + qldbTableName + "(itemKey)");
+            });
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
 
-    private static void addPartnerModels(Collection<Collection<OperationModel>> models, RotatingPartnerStore partnerConfigProvider) {
-        try {
-            String config = partnerConfigProvider.getConfig();
-            Collection<OperationModel> newModels = new HashSet<>();
-            newModels.add(new OperationModel(Type.PARTNER, "singleton", null,
-                    DigestUtils.sha256Hex(config), null));
-            models.add(newModels);
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+    private static void insertIntoQLDB(OperationModel model) { //populates qldb
+        QLDBAuditModel auditModel = new QLDBAuditModel(model.itemType, model.itemKey, model.actionTaken, null,
+                null, null, -1, model.itemHash, model.summary);
+        qldbDriver.execute(txn -> {
+            JsonObject jsonObject = new JsonObject();
+            jsonObject.put("data", auditModel.writeToJson());
+            txn.execute("INSERT INTO " + qldbTableName + " VALUE ?",
+                    ionSys.newLoader().load(auditModel.writeToJson().toString()).get(0));
```

Comment:
again, does this put itemType and itemKey at top level

Reply:
fixed